core/src/main/java/org/stellar/anchor/sep12/Sep12Service.java

```diff
@@ -56,8 +67,19 @@ public Sep12PutCustomerResponse putCustomer(Sep10Jwt token, Sep12PutCustomerRequ
       request.setAccount(token.getAccount());
     }
 
-    return PutCustomerResponse.to(
-        customerIntegration.putCustomer(PutCustomerRequest.from(request)));
+    PutCustomerResponse response =
+        customerIntegration.putCustomer(PutCustomerRequest.from(request));
+
+    // Only publish event if the customer was updated.
+    eventSession.publish(
+        AnchorEvent.builder()
+            .id(UUID.randomUUID().toString())
+            .sep("12")
+            .type(AnchorEvent.Type.CUSTOMER_UPDATED)
+            .customer(CustomerUpdatedResponse.builder().account(request.getAccount()).build())
+            .build());
+
+    return PutCustomerResponse.to(response);
```

Comment:
So is the idea that the flow for the business would be as follows?

1. receives `PUT /customer` callback request, validates and persists KYC information for the customer, and returns
2. consumes an event from the queue that a customer was updated, finds all transactions pending for the customer, and updates their statuses

This feels fairly opinionated to me, because we're effectively suggesting to the business to process the same event twice. The business is made aware of the customer update via the callback and can decide on their own how they want to manage updating the customers' pending transactions. 

I would imagine most anchors will update the customer's pending transactions synchronously in the request/response cycle of the `PUT /customer` call, in which case they don't need to listen for the event. 

If they want to update transactions asynchronously, which is probably the better approach, I guess monitoring the queue is a reasonable way to do that. Given that we're replacing the practice of reading directly from the queue with receiving HTTP webhooks, are we going to update the reference server's implementation?

Reply:
> So is the idea that the flow for the business would be as follows?
> 1. receives PUT /customer callback request, validates and persists KYC information for the customer, and returns
> 2. consumes an event from the queue that a customer was updated, finds all transactions pending for the customer, and updates their statuses

Yes, that's right.

> I would imagine most anchors will update the customer's pending transactions synchronously in the request/response cycle of the PUT /customer call, in which case they don't need to listen for the event.

Hmm... I'd imagine that asking anchors to update their existing integrations would introduce _more_ adoption friction, but I don't have the experience with talking to anchors to say otherwise. If they want to handle deposits synchronously in the `PUT /customer` request path, I don't think emitting an `AnchorEvent` would prevent them from doing so.

> Given that we're replacing the practice of reading directly from the queue with receiving HTTP webhooks, are we going to update the reference server's implementation?

We could. Although, I don't think this is blocking for the SEP-6 launch.

Reply:
Yes it doesn't need to block your progress, just a thought. Lets keep the approach you defined because it doesn't prevent businesses from updating deposit transactions synchronously  👍